src/node_options.cc

```diff
@@ -0,0 +1,335 @@
+#if defined(NODE_HAVE_I18N_SUPPORT)
+const char* NodeOptions::icu_data_dir = nullptr;
+#endif
+
+enum {
+  OPT_NO_DEPRECATION = 1000,
+  OPT_THROW_DEPRECATION,
+  OPT_TRACE_DEPRECATION,
+  OPT_TRACE_SYNC_IO,
+  OPT_V8_OPTIONS,
+  OPT_ABORT_UNCAUGHT,
+  OPT_EXPOSE_INTERNALS,
+  OPT_DEBUG,
+  OPT_DEBUG_BRK,
+  OPT_DEBUG_PORT,
+#if defined(NODE_HAVE_I18N_SUPPORT)
+  OPT_ICU_DATA_DIR
+#endif
+};
+
+static struct option longopts[] = {
+  { "help", no_argument, nullptr, 'h', "show help and usage" },
+  { "version", no_argument, nullptr, 'v', "print io.js version" },
+  { "eval", optional_argument, nullptr, 'e', "evaluate script" },
+  { "print", optional_argument, nullptr, 'p',
+    "evaluate script and print result" },
+  { "interactive", no_argument, nullptr, 'i',
+    "always enter the REPL even if stdin "
+    "does not appear to be a terminal" },
+  { "require", required_argument, nullptr, 'r', "module to preload" },
+  { "no-deprecation", no_argument, nullptr, OPT_NO_DEPRECATION,
+    "silence deprecation warnings" },
+  { "throw-deprecation", no_argument, nullptr, OPT_THROW_DEPRECATION,
+    "throw an exception anytime a deprecated function is used" },
+  { "trace-deprecation", no_argument, nullptr, OPT_TRACE_DEPRECATION,
+    "show stack traces on deprecations" },
+  { "trace-sync-io", no_argument, nullptr, OPT_TRACE_SYNC_IO,
+    "show stake trace when use of sync IO "
+    "is detected after the first tick" },
+  { "v8-options", no_argument, nullptr, OPT_V8_OPTIONS,
+    "print v8 command line options" },
+  { "v8_options", no_argument, nullptr, OPT_V8_OPTIONS,
+    "print v8 command line options" },
+  { "abort-on-uncaught-exception", no_argument, nullptr, OPT_ABORT_UNCAUGHT,
+    "abort on uncaught exception" },
+  { "abort_on_uncaught_exception", no_argument, nullptr, OPT_ABORT_UNCAUGHT,
+    "abort on uncaught exception" },
+  { "expose-internals", no_argument, nullptr, OPT_EXPOSE_INTERNALS,
+    "expose internal modules" },
+  { "expose_internals", no_argument, nullptr, OPT_EXPOSE_INTERNALS,
+    "expose internal modules" },
+  { "debug", optional_argument, nullptr, OPT_DEBUG, "enable debug mode" },
+  { "debug-brk", optional_argument, nullptr, OPT_DEBUG_BRK,
+    "break before starting" },
+  { "debug-port", required_argument, nullptr, OPT_DEBUG_PORT,
+    "specify debug port (defaults to 5858)" },
+#if defined(NODE_HAVE_I18N_SUPPORT)
+  { "icu-data-dir", required_argument, nullptr, OPT_ICU_DATA_DIR },
+#endif
+  { nullptr, 0, nullptr, 0, "" }
+};
+
+void NodeOptions::PrintHelp() {
+  printf("Usage: iojs [options] [ -e script | script.js ] [arguments] \n"
+         "       iojs debug script.js [arguments] \n"
+         "\n"
+         "Options:\n");
+  size_t i;
+  for (i = 0; i < (sizeof longopts / sizeof longopts[0]); i++) {
+    if (longopts[i].name == nullptr)
+      continue;
+    if (longopts[i].val < 1000) {
+      printf("\t-%c, --%-30s %-50s\n",
+             static_cast<char>(longopts[i].val),
```

Comment:
The static_cast is not necessary here, `%c` actually takes an int (which is then converted to a char.)